### ISSUE TITLE
-Wsometimes-uninitialized in arch/powerpc/kexec/file_load_64.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1684]
```
arch/powerpc/kexec/file_load_64.c:1053:11: error: variable 'ret' is used uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
        else if (prop)
                ^~~~
arch/powerpc/kexec/file_load_64.c:1056:9: note: uninitialized use occurs here
        return ret;
              ^~~
arch/powerpc/kexec/file_load_64.c:1053:7: note: remove the 'if' if its condition is always true
        else if (prop)
            ^~~~~~~~~
arch/powerpc/kexec/file_load_64.c:1046:30: note: initialize the variable 'ret' to silence this warning
        int len = 0, fdtlen = 0, ret;
                                    ^
                                    = 0
1 error generated.
```

Reported upstream: https://lore.kernel.org/YvE+d7xcB77GODjc@dev-arch.thelio-3990X/

**Comment by nathanchance** [comment]
Patch submitted: https://lore.kernel.org/20220809054530.907974-1-ruscur@russell.cc/

**Comment by nathanchance** [comment]
v2: https://lore.kernel.org/20220810054331.373761-1-ruscur@russell.cc/

**Comment by nathanchance** [comment]
Appears to be accepted: https://git.kernel.org/powerpc/c/83ee9f23763a432a4077bf20624ee35de87bce99

**Comment by nathanchance** [comment]
Merged in 6.0-rc1: https://git.kernel.org/linus/83ee9f23763a432a4077bf20624ee35de87bce99
